zebra-network/src/address_book.rs

```diff
@@ -435,6 +518,12 @@ impl AddressBook {
         );
 
         if let Some(entry) = self.by_addr.remove(&removed_addr) {
+            // Check if this surplus peer's addr matches that in `most_recent_by_ip`
+            // for this the surplus peer's ip to remove it there as well.
+            if self.should_remove_most_recent_by_ip(entry.addr) {
+                self.most_recent_by_ip.as_mut()?.remove(&entry.addr.ip());
```

Comment:
I think this will return `None` when `most_recent_by_ip` is disabled, even if a peer was actually removed. But this method should return the removed peer.

```suggestion
                if let Some(most_recent_by_ip) = self.most_recent_by_ip.as_mut() {
                    most_recent_by_ip.remove(&entry.addr.ip());
                }
```

Reply:
> I think this will return `None`

It wasn't clear but it's guaranteed to be `Some` if `should_remove_most_recent_by_ip()` is true.